src/math/field_element.rs

```diff
@@ -67,7 +65,7 @@ impl ops::Add<FieldElement> for FieldElement {
     type Output = FieldElement;
 
     fn add(self, a_field_element: FieldElement) -> FieldElement {
-        FieldElement::new((self.value + a_field_element.value) % ORDER).unwrap()
```

Comment:
% ORDER should remain in the operation

Reply:
I think it's fine without the % ORDER. It's the field element that corresponds to the class of the sum of the representatives.

Reply:
> We apply modulo in the constructor instead.

Looks good then